crates/rpc/src/module.rs

```diff
@@ -119,6 +123,46 @@ impl Module {
 
         Ok(self)
     }
+
+    pub fn register_subscription<Subscription, WSAnySubscriptionEvent: 'static + Send + Clone>(
+        mut self,
+        subscription_name: &'static str,
+        subscription_answer_name: &'static str,
+        unsubscription_name: &'static str,
+        subscription: Subscription,
+        ws_broadcast_tx: RPCSender<WSAnySubscriptionEvent>,
+    ) -> anyhow::Result<Self>
+    where
+        Subscription: (Fn(
+                RpcContext,
+                SubscriptionSink,
+                &RPCSender<WSAnySubscriptionEvent>,
+            ) -> SubscriptionResult)
+            + Copy
+            + Send
+            + Sync
+            + 'static,
+    {
+        use anyhow::Context;
+        use jsonrpsee::types::Params;
+
+        metrics::register_counter!("rpc_subscription_calls_total", "subscription" => subscription_name);
```

Comment:
The counter is registered but I don't see it incremented anywhere when an actual subscription is done.

Reply:
In `subscription_callback`?
```rust
metrics::increment_counter!("rpc_subscription_calls_total", "subscription" => subscription_name);
```

Reply:
Sorry, I accepted all the changes first and then reviewed this. Thanks for the suggestion diff!